contracts/BaseRouter.sol

```diff
@@ -618,13 +608,11 @@ abstract contract BaseRouter is Initializable {
     /// @param token Address of the token to change allowance
     /// @param spender Address to change the allowance of
     /// @param amount Amount allowed
-    function _changeAllowance(
-        IERC20 token,
-        address spender,
-        uint256 amount
-    ) internal {
+    function _changeAllowance(IERC20 token, address spender, uint256 amount) internal {
         uint256 currentAllowance = token.allowance(address(this), spender);
-        if (currentAllowance < amount) {
+        // In case `currentAllowance < type(uint256).max / 2` and we want to increase it:
+        // Do nothing (to handle tokens that need reapprovals to 0 and save gas)
```

Comment:
Maybe we should add in the case we do nothing, to actually reset allowance to 0 and bak to max?